# --- SDE-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
#
# Filename: package/.../ncurses/install-no-rm.patch
# Copyright (C) 2006 - 2007 The OpenSDE Project
#
# More information can be found in the files COPYING and README.
#
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- SDE-COPYRIGHT-NOTE-END ---

when deleting ncurses library, as bash depends on it, the world crashes... badly.

--- ./mk-1st.awk.orig	2006-10-17 18:38:56.000000000 +0000
+++ ./mk-1st.awk	2006-10-17 19:14:22.000000000 +0000
@@ -54,8 +54,7 @@
 #
 function symlink(src,dst) {
 		if ( src != dst ) {
-			printf "rm -f %s; ", dst
-			printf "$(LN_S) %s %s; ", src, dst
+			printf "$(LN_S) -nf %s %s; ", src, dst
 		}
 	}
 function rmlink(directory, dst) {
@@ -257,11 +254,11 @@
 					if ( ShlibVer == "cygdll" ) {
 						ovr_name = sprintf("libcurses%s.a", suffix)
 						printf "\t@echo linking %s to %s\n", imp_name, ovr_name
-						printf "\tcd $(DESTDIR)$(libdir) && (rm -f %s; $(LN_S) %s %s; )\n", ovr_name, imp_name, ovr_name
+						printf "\tcd $(DESTDIR)$(libdir) && $(LN_S) -nf %s %s\n", ovr_name, imp_name, ovr_name
 					} else {
 						ovr_name = sprintf("libcurses%s", suffix)
 						printf "\t@echo linking %s to %s\n", end_name, ovr_name
-						printf "\tcd $(DESTDIR)$(libdir) && (rm -f %s; $(LN_S) %s %s; )\n", ovr_name, end_name, ovr_name
+						printf "\tcd $(DESTDIR)$(libdir) && $(LN_S) -nf %s %s\n", ovr_name, end_name, ovr_name
 					}
 				}
 				if ( ldconfig != "" && ldconfig != ":" ) {
@@ -341,8 +338,7 @@
 				if ( overwrite == "yes" && lib_name == "libncurses.a" )
 				{
 					printf "\t@echo linking libcurses.a to libncurses.a\n"
-					printf "\t-@rm -f $(DESTDIR)$(libdir)/libcurses.a\n"
-					printf "\t(cd $(DESTDIR)$(libdir) && $(LN_S) libncurses.a libcurses.a)\n"
+					printf "\t(cd $(DESTDIR)$(libdir) && $(LN_S) -nf libncurses.a libcurses.a)\n"
 				}
 				printf "\t$(RANLIB) $(DESTDIR)$(libdir)/%s\n", lib_name
 				if ( host == "vxworks" )
